core/trino-main/src/main/java/io/trino/server/ServerConfig.java

```diff
@@ -104,4 +121,10 @@ public ServerConfig setQueryInfoUrlTemplate(String queryInfoUrlTemplate)
         this.queryInfoUrlTemplate = Optional.ofNullable(queryInfoUrlTemplate);
         return this;
     }
+
+    public enum CoordinatorShutdownStrategy
+    {
+        CANCEL_RUNNING, // Cancel running queries after grace period has passed
+        ABORT_AFTER_WAITING // Wait for running queries to complete and abort shutdown after grace period has passed
```

Comment:
This name is misleading too, it reads as if the shutdown is _always_ aborted. How about `WAIT_AND_CANCEL_QUERIES` and `WAIT_AND_ABORT_SHUTDOWN`?